frontend/src/lib/keybind.ts

```diff
@@ -0,0 +1,246 @@
+
+  // Get the anchor element and focus it
+  (nextResult.getElementsByClassName('result-main-link')[0] as HTMLElement)?.focus();
+};
+
+/**
+ * Wrapper for {@link navigateResults} that will focus on the next result
+ */
+const focusNextResult = () => navigateResults(1);
+
+/**
+ * Wrapper for {@link navigateResults} that will focus on the previous result
+ */
+const focusPrevResult = () => navigateResults(-1);
+
+/**
+ * Focus the first result (if any)
+ */
+const focusMainResult = () => {
+  const results = document.getElementsByClassName('result-main-link');
+  if (results.length > 0) {
+    (results[0] as HTMLElement).focus();
+  }
+};
+
+/**
+ * Focus the search bar
+ */
+const focusSearchBar = () => {
+  (document.getElementById('searchbar') as HTMLInputElement)?.select();
+};
+
+/**
+ * Scroll to the top of the window and reset focus
+ */
+const scrollToTop = () => {
+  window.scrollTo({ top: 0, behavior: 'smooth' });
+  if (document.activeElement instanceof HTMLElement) {
+    document.activeElement.blur(); // reset focus
+  }
+};
+
+/**
+ * Redirect to the currently focused result
+ *
+ * @param e - The triggering {@link KeyboardEvent}
+ */
+const openResult = (e: KeyboardEvent) => {
+  if ((e.target as HTMLElement).className.includes('result-main-link')) {
+    const resultLink = e.target as HTMLAnchorElement;
+    window.location.href = resultLink.href;
+  } else {
+    const results = document.getElementsByClassName('result-main-link');
+    if (results.length > 0) {
+      window.location.href = (results[0] as HTMLAnchorElement).href;
+    }
+  }
+};
+
+/**
+ * Open the currently focused result in a new tab
+ *
+ * @remarks
+ * Requires pop-ups to be allowed for the window
+ *
+ * @param e - The triggering {@link KeyboardEvent}
+ */
+const openResultInNewTab = (e: KeyboardEvent) => {
+  // Ensure target event is a result url
+  if ((e.target as HTMLElement).className.includes('result-main-link')) {
+    const resultLink = e.target as HTMLAnchorElement;
+    window.open(resultLink.href, '_blank', 'noopener');
+  }
+};
+
+/**
+ * Do a domain search using the domain of the currently focused result
+ *
+ * @param e - The triggering {@link KeyboardEvent}
+ */
+const domainSearch = (e: KeyboardEvent) => {
+  // Ensure target event is a result url
+  if ((e.target as HTMLElement).className.includes('result-main-link')) {
+    const searchBar = document.getElementById('searchbar') as HTMLInputElement;
+    if (searchBar) {
+      // Parse the result url, pull out the hostname then resubmit search
+      const resultLink = new URL((e.target as HTMLAnchorElement)?.href);
+      searchBar.value += ` site:${resultLink.hostname}`;
+      searchBar.form?.submit();
+    }
+  }
+};
+
+/**
+ * Redirect to the `Did you mean:` link (if exists)
+ */
+const goToMisspellLink = () => {
+  const msLink = document.getElementById('misspell-link');
+  if (msLink) window.location.href = (msLink as HTMLAnchorElement).href;
```

Comment:
Should use `Refs` with a binding for the mispell link added to the type instead of `.getElementById` and could then also use a `.open()` method